Source/Core/OrientedBoundingBox.js

```diff
@@ -0,0 +1,332 @@
+
+        // Corner arrangement:
+        //          N/+y
+        //      [0] [1] [2]
+        // W/-x [7]     [3] E/+x
+        //      [6] [5] [4]
+        //          S/-y
+        var perimeterNW = perimeterCartographicScratch[0];
+        var perimeterNC = perimeterCartographicScratch[1];
+        var perimeterNE = perimeterCartographicScratch[2];
+        var perimeterCE = perimeterCartographicScratch[3];
+        var perimeterSE = perimeterCartographicScratch[4];
+        var perimeterSC = perimeterCartographicScratch[5];
+        var perimeterSW = perimeterCartographicScratch[6];
+        var perimeterCW = perimeterCartographicScratch[7];
+
+        var lonCenter = tangentPointCartographic.longitude;
+        var latCenter = tangentPointCartographic.latitude;
+        perimeterSW.latitude = perimeterSC.latitude = perimeterSE.latitude = rectangle.south;
+        perimeterCW.latitude = perimeterCE.latitude = latCenter;
+        perimeterNW.latitude = perimeterNC.latitude = perimeterNE.latitude = rectangle.north;
+        perimeterSW.longitude = perimeterCW.longitude = perimeterNW.longitude = rectangle.west;
+        perimeterSC.longitude = perimeterNC.longitude = lonCenter;
+        perimeterSE.longitude = perimeterCE.longitude = perimeterNE.longitude = rectangle.east;
+
+        // Compute XY extents using the rectangle at maximum height
+        perimeterNE.height = perimeterNW.height = perimeterSE.height = perimeterSW.height = perimeterNC.height = perimeterSC.height = maximumHeight;
+
+        ellipsoid.cartographicArrayToCartesianArray(perimeterCartographicScratch, perimeterCartesianScratch);
+        tangentPlane.projectPointsToNearestOnPlane(perimeterCartesianScratch, perimeterProjectedScratch);
+        // See the `perimeterXX` definitions above for what these are
+        var minX = Math.min(perimeterProjectedScratch[6].x, perimeterProjectedScratch[7].x, perimeterProjectedScratch[0].x);
+        var maxX = Math.max(perimeterProjectedScratch[2].x, perimeterProjectedScratch[3].x, perimeterProjectedScratch[4].x);
+        var minY = Math.min(perimeterProjectedScratch[4].y, perimeterProjectedScratch[6].y, perimeterProjectedScratch[6].y);
+        var maxY = Math.max(perimeterProjectedScratch[0].y, perimeterProjectedScratch[1].y, perimeterProjectedScratch[2].y);
+
+        // Compute minimum Z using the rectangle at minimum height
+        perimeterNE.height = perimeterNW.height = perimeterSE.height = perimeterSW.height = minimumHeight;
+        ellipsoid.cartographicArrayToCartesianArray(perimeterCartographicScratch, perimeterCartesianScratch);
+        var minZ = Math.min(Plane.getPointDistance(plane, perimeterCartesianScratch[0]),
+                            Plane.getPointDistance(plane, perimeterCartesianScratch[1]),
+                            Plane.getPointDistance(plane, perimeterCartesianScratch[2]),
+                            Plane.getPointDistance(plane, perimeterCartesianScratch[3]));
```

Comment:
This is a change that I meant to amend on but didn't add before stopping work on Friday. It's meant to be 0, 2, 4, 6 (the corner points at minimum height).